h/services/list_groups.py

```diff
@@ -44,24 +44,38 @@ def all_groups(self, user=None, authority=None, document_uri=None):
         """
         Return a list of groups relevant to this session/profile (i.e. user).
 
+        Return a list of groups filtered on user and authority. All open
+        groups matching the authority will be included.
+        """
+        all_open_groups = self._open_groups(user, authority)
+        private_groups = self._private_groups(user)
+
+        return all_open_groups + private_groups
+
+    def request_groups(self, authority, user=None, document_uri=None):
```

Comment:
I have bifurcated the service based on some good feedback from @seanh the other day—there is (or potentially is) a difference between "get me all groups for the user/session" versus "get me all groups for the current request context" and this new method name reflects the latter. This is the method that is going to do the filtering based on `document_uri`.

Reply:
I've alerted @hmstepanek (currently the only other dev-consumer of `ListGroupsService.all_groups`) that I may opt to rename that method at some point in the future to reflect its user-session-ness, but I will make sure everyone is kept in the loop.

Reply:
I like! QQ: if all_groups is no longer getting filtered by uri should we remove the uri param?

Reply:
@hmstepanek Brilliantly observed. Yes, this is on the horizon, likely.